src/server/api/routes/users/v2/follow.js

```diff
@@ -41,12 +47,33 @@ router.post(
         res.status(400).send('INVALID_CREATOR_ADDRESS');
         return;
       }
-      await walletUserCollection.doc(user).update({
-        followers: FieldValue.arrayUnion(creator),
+      await db.runTransaction(async t => {
+        const userRef = walletUserCollection.doc(user);
+        const userDoc = await t.get(userRef);
+        const { email, emailUnconfirmed } = userDoc.data();
+        if (!email) {
+          if (emailUnconfirmed) {
+            throw new Error('EMAIL_UNCONFIRMED');
+          } else {
+            throw new Error('EMAIL_NOT_SET_YET');
+          }
+        }
+        await t.update(userRef, {
```

Comment:
note that t.update is not an async call

Reply:
I guess unless `t.update()` is the last line in a `runTransaction()`, we should add an `await` for it?
https://firebase.google.com/docs/firestore/manage-data/transactions#passing_information_out_of_transactions